.gitlab-ci.yml

```diff
@@ -1034,7 +1034,6 @@ workflow:
   - changes:
       paths: *security_agent_change_paths
       compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
-    when: on_success
```

Comment:
While I agree this is unnecessary because `when: on_success` is the default behaviour, I'm not sure this changes (fix) anything. It seems on the pipeline you quoted there was a cancellation that led the cleanup jobs to be skipped, wdyt?

Reply:
I think [this](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/30908527) is the only pipeline with skipped cleanup jobs and in this case that's correct, as the setup-env jobs don't run either.